tests/test_webassets.py

```diff
@@ -0,0 +1,320 @@
+        generator = self.get_generators()
+        self.assertTrue(generator.env.assets_environment.debug)
+
+    def test_webassets_bundles(self):
+        """ensure webassets.Bundles are passed to the webassets module"""
+        bundle_name = 'test_bundle_name'
+        bundle_args = ('test', 'arguments', 'for', 'webassets')
+        bundle_kwargs = {'output': 'test_bundle_output', 'filters': ['libsass']}
+        generator = self.get_generators({
+            'WEBASSETS_BUNDLES': [(bundle_name, bundle_args, bundle_kwargs)]})
+
+        ## TODO: Super Hacky
+        bundles = generator.env.assets_environment._named_bundles
+        self.assertIn(bundle_name, bundles)
+
+        # ensure all arguments are given
+        test_bundle = bundles[bundle_name]
+        for argument in bundle_args:
+            self.assertIn(argument, test_bundle.contents)
+
+        # ensure the libsass filter is used
+        from webassets.filter.libsass import LibSass
+        self.assertIn(LibSass(), test_bundle.filters)
+
+    def test_asset_bundles(self):
+        """ensure ASSET_BUNDLES are passed to the webassets module"""
+        bundle_name = 'asset_test_bundle_name'
+        bundle_args = ('asset', 'arguments', 'for', 'webassets')
+        bundle_kwargs = {'output': 'asset_bundle_output', 'filters': ['libsass']}
+        generator = self.get_generators({
+            'ASSET_BUNDLES': [(bundle_name, bundle_args, bundle_kwargs)]})
+
+        ## TODO: Super Hacky
+        bundles = generator.env.assets_environment._named_bundles
+        self.assertIn(bundle_name, bundles)
+
+        # ensure all arguments are given
+        test_bundle = bundles[bundle_name]
+        for argument in bundle_args:
+            self.assertIn(argument, test_bundle.contents)
+
+        # ensure the libsass filter is used
+        from webassets.filter.libsass import LibSass
+        self.assertIn(LibSass(), test_bundle.filters)
+
+    def test_webassets_source_paths(self):
+        """ensure WEBASSETS_SOURCE_PATHS is passed to the webassets module"""
+        source_paths = ['some', 'random', 'source', 'paths/for/webassets']
+        generator = self.get_generators({
+            'WEBASSETS_SOURCE_PATHS': source_paths})
+
+        paths = generator.env.assets_environment.load_path
+
+        for path in self.settings['THEME_STATIC_PATHS'] + source_paths:
+            self.assertIn(str(Path(generator.theme, path)), paths)
+
+    def test_assets_source_paths(self):
+        """ensure ASSET_SOURCE_PATHS is passed to the webassets module"""
+        source_paths = ['random', 'source', 'paths/for/webassets']
+        generator = self.get_generators({
+            'ASSET_SOURCE_PATHS': source_paths})
+
+        paths = generator.env.assets_environment.load_path
+
+        for path in self.settings['THEME_STATIC_PATHS'] + source_paths:
+            self.assertIn(str(Path(generator.theme, path)), paths)
+
+
+class TestDepricationDate(unittest.TestCase):
```

Comment:
Ahaha, that's a fun one! 😄 Great way to keep a reasonable time-frame for the migration window! 😎